### PR TITLE
Bdog 2360

### PR DESCRIPTION
**Pull request by Shnick**

### CODE REVIEW

File: app/uk/gov/hmrc/servicecommissioningstatus/connectors/TeamsAndRepositoriesConnector.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  sealed trait Tag {def asString: String}
+
+  object Tag {
+    case object AdminFrontend extends Tag { def asString = "AdminFrontend" }
+    case object Api           extends Tag { def asString = "Api"           }
+    case object Stub          extends Tag { def asString = "Stub"          }
+
+    val values = Set(AdminFrontend, Api, Stub)
+
+    def parse(s: String): Either[String, Tag] =
```
Comment (chotai) [inline app/uk/gov/hmrc/servicecommissioningstatus/connectors/TeamsAndRepositoriesConnector.scala:58]: This is a duplication of Line 37 buy the looks of it

File: app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala [hunk 2]
```diff
@@ -44,20 +45,25 @@ class StatusCheckService @Inject()(
   import Check.{SimpleCheck, EnvCheck}
   def commissioningStatusChecks(serviceName: String)(implicit hc: HeaderCarrier): Future[List[Check]] =
     for {
+      oRepo           <- teamsAndReposConnector.findRepo(serviceName)
+      isFrontend      =  oRepo.map(_.serviceType) == Some(TeamsAndRepositoriesConnector.ServiceType.Frontend)
```
Comment (chotai) [inline app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala:49]: This comparison looks odd - can we use exists?

File: app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala [hunk 3]
```diff
@@ -105,19 +112,22 @@ class StatusCheckService @Inject()(
...
       .flatMap(_.routes.map(_.ruleConfigurationUrl).headOption) match {
         case Some(e) => Right(Check.Present(e))
         case None    => Left(Check.Missing(s"https://github.com/hmrc/mdtp-frontend-routes/${env.asString}"))
       }
 
+  private def checkAdminFrontendRouteForEnv(adminFrontendRoutes: Seq[ServiceConfigsConnector.AdminFrontendRoute], env: Environment): Check.Result =
+    adminFrontendRoutes.headOption match {
+      case Some(e)
+        if e.allow.get(env.asString).isDefined
+             => Right(Check.Present(e.location))
```
Comment (chotai) [inline app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala:127]: what if it isn't defined?

File: app/uk/gov/hmrc/servicecommissioningstatus/connectors/TeamsAndRepositoriesConnector.scala (hunk 1 above)
Comment (Shnick) [reply app/uk/gov/hmrc/servicecommissioningstatus/connectors/TeamsAndRepositoriesConnector.scala:58]: This is a copy of how we treat types in our other services such as Environment. It's easier to lift and re-use as we can't create a shared library.

File: app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala (hunk 2 above)
Comment (Shnick) [reply app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala:49]: yeah sure

File: app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala (hunk 3 above)
Comment (Shnick) [reply app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala:127]: handled by `case _ => Left(...) ` below

File: app/uk/gov/hmrc/servicecommissioningstatus/connectors/TeamsAndRepositoriesConnector.scala (hunk 1 above)
Comment (Shnick) [reply app/uk/gov/hmrc/servicecommissioningstatus/connectors/TeamsAndRepositoriesConnector.scala]: Made `StringType` trait to reduce duplicate code

**Review by colin-lamed — APPROVED** [review]

**Pull request merged**